### PR TITLE
Remove legacy file based block dag store

### PR DESCRIPTION
**Pull request by nzpr**
## Overview
File based dag store (along with finalised block store) is not needed anymore once KV store is introduced. Migration from file to KV store can be done by running previous releases, e.g. https://github.com/rchain/rchain/releases/tag/v0.10.1, process is automatic and happen on startup. In future releases with this PR merged, file storage wont be supported anymore and migration won't be possible.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/NodeRuntime.scala [hunk 1]
```diff
@@ -770,12 +751,6 @@ object NodeRuntime {
...
           blockMetadataDb   <- rnodeStoreManager.store("block-metadata")
           dagStorageIsEmpty = blockMetadataDb.iterate(_.isEmpty)
-          oldStorageExists  = Sync[F].delay(Files.exists(dagConfig.blockMetadataLogPath))
-          shouldMigrate     <- dagStorageIsEmpty &&^ oldStorageExists
-          // TODO: remove `dagConfig`, it's not used anymore (after migration)
-          _ <- BlockDagKeyValueStorage
-                .importFromFileStorage(rnodeStoreManager, dagConfig)
-                .whenA(shouldMigrate)
           // Create DAG store
           dagStorage <- BlockDagKeyValueStorage.create[F](rnodeStoreManager)
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: This is the only line that is needed.
```scala
// Create DAG store
blockDagStorage <- BlockDagKeyValueStorage.create[F](rnodeStoreManager)
```

**Review by tgrospic — COMMENTED** [review]
Please check `PersistentIndex` and all places where is used for removal also.
https://github.com/rchain/rchain/blob/6d1cd6230d74f3ad8beb7f809cde272bee4fdd38/block-storage/src/main/scala/coop/rchain/blockstorage/dag/PersistentIndex.scala#L22

Also some of the errors will become unused that can be removed together with these helpers.
https://github.com/rchain/rchain/blob/0801b6be616872c7e7c40c26e14c789d76328f14/block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala#L28-L62

File: block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala [hunk 2]
```diff
@@ -1,58 +1,32 @@
...
-final case object DataLookupIsCorrupted                      extends StorageError
-final case object InvalidBlocksIsCorrupted                   extends StorageError
-final case object BlockHashesByDeployLogIsCorrupted          extends StorageError
-final case object LastFinalizedBlockIsCorrupted              extends StorageError
+case object LatestMessagesLogIsCorrupted                     extends StorageError
+case object LastFinalizedBlockIsCorrupted                    extends StorageError
 
 object StorageError {
   type StorageErr[A]        = Either[StorageError, A]
   type StorageErrT[F[_], A] = EitherT[F, StorageError, A]
```
Comment (tgrospic) [inline block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala]: This is also not used.

File: block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala [hunk 3]
```diff
@@ -1,58 +1,32 @@
...
         "Latest messages log is corrupted"
-      case DataLookupIsCorrupted =>
-        "Data lookup log is corrupted"
-      case InvalidBlocksIsCorrupted =>
-        "Invalid blocks log is corrupted"
-      case BlockHashesByDeployLogIsCorrupted =>
-        "Block hashes by deploy log is corrupted"
       case LastFinalizedBlockIsCorrupted =>
         "Last finalized block file is corrupted"
     }
```
Comment (tgrospic) [inline block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala]: And the `errorMessage` also, which also makes `StorageErrorToMessage` unused.

File: block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala (hunk 2 above)
Comment (nzpr) [reply block-storage/src/main/scala/coop/rchain/blockstorage/errors.scala]: This is still used in file block store isn't it?

**Comment by nzpr** [comment]
Also removed file based last finalised block storage. I think other stuff in `errors.scala` is still used by file block store

**Review by tgrospic — APPROVED** [review]
_The great purge._ :smiley: 

I've added one small commit, few more removed errors.

**Comment by tgrospic** [comment]
I've added one more commit to remove _checkpoint_ and _close_ methods from BlockDagStorage.

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by nzpr** [comment]
bors merge

**Pull request merged**